src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace System.Net.NetworkInformation
+{
+    // Linux implementation of NetworkChange
+    public class NetworkChange
+    {
+        private static NetworkAddressChangedEventHandler s_addressChangedSubscribers;
+        private static int s_socket = 0;
+        private static readonly object s_lockObj = new object();
+
+        public static event NetworkAddressChangedEventHandler NetworkAddressChanged
+        {
+            add
+            {
+                lock (s_lockObj)
+                {
+                    if (s_socket == 0)
+                    {
+                        CreateSocket();
+                    }
+
+                    s_addressChangedSubscribers += value;
+                }
+            }
+            remove
+            {
+                lock (s_lockObj)
+                {
+                    if (s_addressChangedSubscribers == null)
+                    {
+                        Debug.Assert(s_socket == 0);
+                        return;
+                    }
+
+                    s_addressChangedSubscribers -= value;
+                    if (s_addressChangedSubscribers == null)
+                    {
+                        CloseSocket();
+                    }
+                }
+            }
+        }
+
+        private static void CreateSocket()
+        {
+            Debug.Assert(s_socket == 0);
+            int newSocket;
+            Interop.Error result = Interop.Sys.CreateNetworkChangeListenerSocket(out newSocket);
+            if (result != Interop.Error.SUCCESS)
+            {
+                string message = Interop.Sys.GetLastErrorInfo().GetErrorMessage();
+                throw new NetworkInformationException(message);
+            }
+
+            s_socket = newSocket;
+            Task.Run(() => LoopReadSocket(s_socket));
```

Comment:
We expect this to be a potentially very-long running work item, right?  I'd suggest changing it to this:
```C#
Task.Factory.StartNew(() => LoopReadSocket(s_socket), 
    CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default);
```
That LongRunning flag will help to avoid taking up a thread pool thread for it.